libcontainer/specconv/spec_linux.go

```diff
@@ -508,6 +508,15 @@ func stringToDeviceRune(s string) (rune, error) {
 	}
 }
 
+func deviceIsExist(dev []*configs.Device, path string) bool {
+	for _, d := range dev {
+		if d.Path == path {
```

Comment:
I'd prefer if this check did something more like `reflect.DeepEqual(d, <the device being checked in the caller>)`. My main concern about this patch is whether we're going to miss something if a user added additional permissions to the device that we implicitly add.

However, I actually don't like that we have pre-configured devices *at all* because it feels like working around a spec issue...

Reply:
@keloyang You didn't get @cyphar 's point, you should do more check than just `d.Path` to confirm the device is exist, and the critical elements for a device should be `type`, `major` and `minor`.

BTW, I agree we should not have duplicated devices for runc to create, but why would this fail with user ns enabled? Doesn't runc just do bind mount and the former mount point would be overlaid by the later one?